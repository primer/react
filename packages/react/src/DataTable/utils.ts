// Utility type to generate an array with a given length. Each member of the
// array type is the length of the array
type ArrayOfLength<Length extends number, SizedArray extends Array<unknown> = []> = SizedArray['length'] extends Length
  ? SizedArray
  : ArrayOfLength<Length, [...SizedArray, SizedArray['length']]>

// A union of the valid lengths of an Array that we'll support in `ObjectPaths`
// below. This is to prevent unbounded arrays from polluting expected types
type MaxLength = ArrayOfLength<10>[number]

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type ArrayIndex<A extends ReadonlyArray<any>, Keys extends number = never> = A extends readonly []
  ? Keys
  : // eslint-disable-next-line @typescript-eslint/no-unused-vars
    A extends readonly [infer _, ...infer Tail]
    ? ArrayIndex<Tail, Keys | Tail['length']>
    : Keys

// Check if the given type is within the bounds set by `MaxLength`
// eslint-disable-next-line @typescript-eslint/no-explicit-any
type ArrayWithinBounds<T> = T extends ReadonlyArray<any> & {length: infer Length}
  ? Length extends MaxLength
    ? T
    : never
  : never

// Get all valid "paths" for an object. This is useful for APIs which require
// a string input that must refer to a valid type for the data being used.
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type ObjectPaths<T> = T extends readonly any[] & ArrayWithinBounds<T>
  ? `${ArrayIndex<T>}` | PrefixPath<T, ArrayIndex<T>>
  : // eslint-disable-next-line @typescript-eslint/no-explicit-any
    T extends any[]
    ? never & 'Unable to determine keys of potentially boundless array'
    : T extends Date
      ? never
      : T extends object
        ? Extract<keyof T, string | number> | PrefixPath<T, Extract<keyof T, string | number>>
        : never

type PrefixPath<T, Prefix> =
  Prefix extends Extract<keyof T, number | string> ? `${Prefix}.${ObjectPaths<T[Prefix]>}` : never

// Get the value of a given path within an object
export type ObjectPathValue<ObjectType extends object, Path extends string | number> =
  ObjectType extends Record<
    string | number,
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    any
  >
    ? Path extends `${infer Key}.${infer NestedPath}`
      ? ObjectPathValue<ObjectType[Key], NestedPath>
      : ObjectType[Path]
    : never
