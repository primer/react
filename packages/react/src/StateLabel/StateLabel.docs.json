{
  "id": "state_label",
  "name": "StateLabel",
  "status": "alpha",
  "a11yReviewed": false,
  "stories": [],
  "importPath": "@primer/react",
  "props": [
    {
      "name": "variant",
      "type": "'small' | 'normal'",
      "defaultValue": "'normal'"
    },
    {
      "name": "status",
      "type": "'issueOpened' | 'issueClosed' | 'issueClosedNotPlanned' | 'pullOpened' | 'pullClosed' | 'pullMerged' | 'draft' | 'issueDraft' | 'unavailable'",
      "required": true
    },
    {
      "name": "sx",
      "type": "SystemStyleObject"
    }
  ],
  "subcomponents": []
}
