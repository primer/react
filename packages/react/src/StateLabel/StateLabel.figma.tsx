import React from 'react'
import {StateLabel} from '../../src'
import figma from '@figma/code-connect'

figma.connect(
  StateLabel,
  'https://www.figma.com/design/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=18959-64962&t=gQF1OJTEGB3vTxPz-4',
  {
    props: {
      size: figma.enum('size', {
        small: 'small',
        normal: 'normal',
      }),
      status: figma.enum('status', {
        draft: 'issueDraft',
        open: 'issueOpened',
        closed: 'issueClosed',
        unavailable: 'unavailable',
      }),
      text: figma.textContent('Label'),
    },
    variant: {variant: 'issue'},
    example: ({text, size, status}) => (
      <StateLabel variant={size} status={status}>
        {text}
      </StateLabel>
    ),
  },
)

figma.connect(
  StateLabel,
  'https://www.figma.com/design/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=18959-64962&t=gQF1OJTEGB3vTxPz-4',
  {
    props: {
      size: figma.enum('size', {
        small: 'small',
        normal: 'normal',
      }),
      status: figma.enum('status', {
        draft: 'draft',
        open: 'pullOpened',
        closed: 'pullClosed',
        merged: 'pullMerged',
        queued: 'pullQueued',
        unavailable: 'unavailable',
      }),
      text: figma.textContent('Label'),
    },
    variant: {variant: 'pull request'},
    example: ({text, size, status}) => (
      <StateLabel variant={size} status={status}>
        {text}
      </StateLabel>
    ),
  },
)
