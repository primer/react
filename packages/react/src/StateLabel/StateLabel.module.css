.StateLabel {
  display: inline-flex;
  align-items: center;
  font-weight: var(--base-text-weight-semibold);
  /* stylelint-disable-next-line primer/typography */
  line-height: 16px;
  color: var(--fgColor-onEmphasis);
  text-align: center;
  border-radius: var(--borderRadius-full);
}

/* Size variants */
.StateLabel:where([data-variant='small']) {
  padding: var(--base-size-4) var(--base-size-8);
  font-size: var(--text-body-size-small);
}

.StateLabel:where([data-variant='normal']) {
  padding: var(--base-size-8) var(--base-size-12);
  font-size: var(--text-body-size-medium);
}

/* Status color variants */
.StateLabel:where([data-status='issueClosed']) {
  background-color: var(--bgColor-done-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-done-emphasis, transparent);
}

.StateLabel:where([data-status='issueClosedNotPlanned']) {
  background-color: var(--bgColor-neutral-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-neutral-emphasis, transparent);
}

.StateLabel:where([data-status='pullClosed']) {
  background-color: var(--bgColor-closed-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-closed-emphasis, transparent);
}

.StateLabel:where([data-status='pullMerged']) {
  background-color: var(--bgColor-done-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-done-emphasis, transparent);
}

.StateLabel:where([data-status='pullQueued']) {
  background-color: var(--bgColor-attention-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-attention-emphasis, transparent);
}

.StateLabel:where([data-status='issueOpened']) {
  background-color: var(--bgColor-open-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-open-emphasis, transparent);
}

.StateLabel:where([data-status='pullOpened']) {
  background-color: var(--bgColor-open-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-open-emphasis, transparent);
}

.StateLabel:where([data-status='draft']) {
  /* stylelint-disable-next-line primer/colors */
  background-color: var(--bgColor-draft-emphasis, var(--bgColor-neutral-emphasis));
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px)
    var(--borderColor-draft-emphasis, var(--borderColor-neutral-emphasis, transparent));
}

.StateLabel:where([data-status='issueDraft']) {
  /* stylelint-disable-next-line primer/colors */
  background-color: var(--bgColor-draft-emphasis, var(--bgColor-neutral-emphasis));
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px)
    var(--borderColor-draft-emphasis, var(--borderColor-neutral-emphasis, transparent));
}

.StateLabel:where([data-status='unavailable']) {
  background-color: var(--bgColor-neutral-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-neutral-emphasis, transparent);
}

.StateLabel:where([data-status='open']) {
  background-color: var(--bgColor-open-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-open-emphasis, transparent);
}

.StateLabel:where([data-status='closed']) {
  background-color: var(--bgColor-done-emphasis);
  color: var(--fgColor-onEmphasis);
  box-shadow: var(--boxShadow-thin, inset 0 0 0 1px) var(--borderColor-done-emphasis, transparent);
}

.Icon {
  margin-right: var(--base-size-4);
}

.Icon:where([data-variant='small']) {
  width: 1em;
}
