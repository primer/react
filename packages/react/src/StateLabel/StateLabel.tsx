import {
  GitMergeIcon,
  GitPullRequestIcon,
  GitPullRequestClosedIcon,
  GitPullRequestDraftIcon,
  IssueClosedIcon,
  SkipIcon,
  IssueDraftIcon,
  IssueOpenedIcon,
  QuestionIcon,
  GitMergeQueueIcon,
  AlertIcon,
} from '@primer/octicons-react'
import React from 'react'
import styled from 'styled-components'
import {variant} from 'styled-system'
import {get} from '../constants'
import Octicon from '../Octicon'
import type {SxProp} from '../sx'
import sx from '../sx'
import type {ComponentProps} from '../utils/types'

const octiconMap = {
  issueOpened: IssueOpenedIcon,
  pullOpened: GitPullRequestIcon,
  issueClosed: IssueClosedIcon,
  issueClosedNotPlanned: SkipIcon,
  pullClosed: GitPullRequestClosedIcon,
  pullMerged: GitMergeIcon,
  draft: GitPullRequestDraftIcon,
  issueDraft: IssueDraftIcon,
  pullQueued: GitMergeQueueIcon,
  unavailable: AlertIcon,
}

const colorVariants = variant({
  prop: 'status',
  variants: {
    issueClosed: {
      backgroundColor: 'done.emphasis',
      color: 'fg.onEmphasis',
    },
    issueClosedNotPlanned: {
      backgroundColor: 'neutral.emphasis',
      color: 'fg.onEmphasis',
    },
    pullClosed: {
      backgroundColor: 'closed.emphasis',
      color: 'fg.onEmphasis',
    },
    pullMerged: {
      backgroundColor: 'done.emphasis',
      color: 'fg.onEmphasis',
    },
    pullQueued: {
      backgroundColor: 'attention.emphasis',
      color: 'fg.onEmphasis',
    },
    issueOpened: {
      backgroundColor: 'open.emphasis',
      color: 'fg.onEmphasis',
    },
    pullOpened: {
      backgroundColor: 'open.emphasis',
      color: 'fg.onEmphasis',
    },
    draft: {
      backgroundColor: 'neutral.emphasis',
      color: 'fg.onEmphasis',
    },
    issueDraft: {
      backgroundColor: 'neutral.emphasis',
      color: 'fg.onEmphasis',
    },
    unavailable: {
      backgroundColor: 'neutral.emphasis',
      color: 'fg.onEmphasis',
    },
  },
})

const sizeVariants = variant({
  prop: 'variant',
  variants: {
    small: {
      paddingX: 2,
      paddingY: 1,
      fontSize: 0,
    },
    normal: {
      paddingX: '12px',
      paddingY: 2,
      fontSize: 1,
    },
  },
})

type StyledStateLabelBaseProps = {
  variant?: 'small' | 'normal'
  status: keyof typeof octiconMap
} & SxProp

const StateLabelBase = styled.span<StyledStateLabelBaseProps>`
  display: inline-flex;
  align-items: center;
  font-weight: ${get('fontWeights.bold')};
  line-height: 16px;
  color: ${get('colors.canvas.default')};
  text-align: center;
  border-radius: ${get('radii.3')};
  ${colorVariants};
  ${sizeVariants};
  ${sx};
`

export type StateLabelProps = ComponentProps<typeof StateLabelBase>

function StateLabel({children, status, variant: variantProp = 'normal', ...rest}: StateLabelProps) {
  const octiconProps = variantProp === 'small' ? {width: '1em'} : {}
  return (
    <StateLabelBase {...rest} variant={variantProp} status={status}>
      {/* eslint-disable-next-line @typescript-eslint/no-unnecessary-condition */}
      {status && <Octicon {...octiconProps} icon={octiconMap[status] || QuestionIcon} sx={{mr: 1}} />}
      {children}
    </StateLabelBase>
  )
}

export default StateLabel
