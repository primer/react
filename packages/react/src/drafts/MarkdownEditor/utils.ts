import {isMacOS} from '@primer/behaviors/utils'

export const getSelectedLineRange = (textarea: HTMLTextAreaElement): [number, number] => {
  // Subtract one from the caret position so the newline found is not the one _at_ the caret position
  // then add one because we don't want to include the found newline. Also changes -1 (not found) result to 0
  const start = textarea.value.lastIndexOf('\n', textarea.selectionStart - 1) + 1

  // activeLineEnd will be the index of the next newline inclusive, which works because slice is last-index exclusive
  let end = textarea.value.indexOf('\n', textarea.selectionEnd)
  if (end === -1) end = textarea.value.length

  return [start, end]
}

export const markdownComment = (text: string) => `<!-- ${text.replaceAll('--', '\\-\\-')} -->`

export const markdownLink = (text: string, url: string) =>
  `[${text.replaceAll('[', '\\[').replaceAll(']', '\\]')}](${url.replaceAll('(', '\\(').replaceAll(')', '\\)')})`

export const markdownImage = (altText: string, url: string) => `!${markdownLink(altText, url)}`

export const isModifierKey = (event: KeyboardEvent | React.KeyboardEvent<unknown>) =>
  isMacOS() ? event.metaKey : event.ctrlKey
