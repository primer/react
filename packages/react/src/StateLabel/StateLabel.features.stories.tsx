import React from 'react'
import type {Meta} from '@storybook/react'
import type {ComponentProps} from '../utils/types'
import StateLabel from './StateLabel'

export default {
  title: 'Components/StateLabel/Features',
  component: StateLabel,
} as Meta<ComponentProps<typeof StateLabel>>

export const IssueOpened = () => <StateLabel status="issueOpened">Open</StateLabel>
export const IssueClosed = () => <StateLabel status="issueClosed">Closed</StateLabel>
export const IssueClosedNotPlanned = () => <StateLabel status="issueClosedNotPlanned">Closed</StateLabel>
export const IssueDraft = () => <StateLabel status="issueDraft">Draft</StateLabel>

export const PullOpened = () => <StateLabel status="pullOpened">Open</StateLabel>
export const PullClosed = () => <StateLabel status="pullClosed">Closed</StateLabel>
export const PullMerged = () => <StateLabel status="pullMerged">Merged</StateLabel>
export const Queued = () => <StateLabel status="pullQueued">Queued</StateLabel>
export const Draft = () => <StateLabel status="draft">Draft</StateLabel>
export const Unavailable = () => <StateLabel status="unavailable">Unavailable</StateLabel>

export const Small = () => (
  <StateLabel status="issueOpened" variant="small">
    Open
  </StateLabel>
)
