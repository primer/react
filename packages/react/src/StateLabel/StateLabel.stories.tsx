import React from 'react'
import type {Meta, StoryFn} from '@storybook/react'
import type {ComponentProps} from '../utils/types'
import StateLabel from './StateLabel'

export default {
  title: 'Components/StateLabel',
  component: StateLabel,
} as Meta<ComponentProps<typeof StateLabel>>

export const Default = () => <StateLabel status="issueOpened">Open</StateLabel>

export const Playground: StoryFn<ComponentProps<typeof StateLabel>> = args => <StateLabel {...args}>Label</StateLabel>

Playground.args = {
  status: 'issueOpened',
}

Playground.argTypes = {
  ref: {
    controls: false,
    table: {
      disable: true,
    },
  },
  sx: {
    controls: false,
    table: {
      disable: true,
    },
  },
  theme: {
    controls: false,
    table: {
      disable: true,
    },
  },
}
