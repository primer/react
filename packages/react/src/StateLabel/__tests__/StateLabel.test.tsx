import React from 'react'
import StateLabel from '../StateLabel'
import {render, behavesAsComponent, checkExports} from '../../utils/testing'
import {render as HTMLRender} from '@testing-library/react'
import axe from 'axe-core'

describe('StateLabel', () => {
  behavesAsComponent({
    Component: StateLabel,
    toRender: () => <StateLabel status="issueOpened">Open</StateLabel>,
    options: {
      // Rendering a Octicon seems to break getComputedStyles, which
      // the sx prop implementation test uses to make sure the prop is working correctly.
      // Despite my best efforts, I cannot figure out why this is happening. So,
      // unfortunately, we will simply skip this test.
      skipSx: true,
    },
  })

  checkExports('StateLabel', {
    default: StateLabel,
  })

  it('should have no axe violations', async () => {
    const {container} = HTMLRender(<StateLabel status="issueOpened" />)
    const results = await axe.run(container)
    expect(results).toHaveNoViolations()
  })

  it('respects the status prop', () => {
    expect(render(<StateLabel status="issueOpened" />)).toMatchSnapshot()
    expect(render(<StateLabel status="issueClosed" />)).toMatchSnapshot()
    expect(render(<StateLabel status="issueClosedNotPlanned" />)).toMatchSnapshot()
    expect(render(<StateLabel status="pullMerged" />)).toMatchSnapshot()
    expect(render(<StateLabel status="pullQueued" />)).toMatchSnapshot()
  })

  it('respects the variant prop', () => {
    expect(render(<StateLabel variant="small" status="issueOpened" />)).toMatchSnapshot()
    expect(render(<StateLabel variant="normal" status="issueOpened" />)).toMatchSnapshot()
  })

  it('renders children', () => {
    expect(render(<StateLabel status="issueOpened">hi</StateLabel>)).toMatchSnapshot()
  })
})
