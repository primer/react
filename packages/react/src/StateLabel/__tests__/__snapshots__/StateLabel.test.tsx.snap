// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateLabel renders children 1`] = `
.c1 {
  margin-right: 4px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  line-height: 16px;
  color: var(--bgColor-default,var(--color-canvas-default,#ffffff));
  text-align: center;
  border-radius: 100px;
  background-color: var(--bgColor-open-emphasis,var(--color-open-emphasis,#1f883d));
  color: var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));
  box-shadow: var(--boxShadow-thin,inset 0 0 0 1px) var(--borderColor-open-emphasis,transparent);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 14px;
}

<span
  className="c0"
>
  <svg
    aria-label="Issue"
    className="c1"
    fill="currentColor"
    focusable="false"
    height={16}
    role="img"
    style={
      {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  >
    <path
      d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"
    />
    <path
      d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z"
    />
  </svg>
  hi
</span>
`;

exports[`StateLabel respects the status prop 1`] = `
.c1 {
  margin-right: 4px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  line-height: 16px;
  color: var(--bgColor-default,var(--color-canvas-default,#ffffff));
  text-align: center;
  border-radius: 100px;
  background-color: var(--bgColor-open-emphasis,var(--color-open-emphasis,#1f883d));
  color: var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));
  box-shadow: var(--boxShadow-thin,inset 0 0 0 1px) var(--borderColor-open-emphasis,transparent);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 14px;
}

<span
  className="c0"
>
  <svg
    aria-label="Issue"
    className="c1"
    fill="currentColor"
    focusable="false"
    height={16}
    role="img"
    style={
      {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  >
    <path
      d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"
    />
    <path
      d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z"
    />
  </svg>
</span>
`;

exports[`StateLabel respects the status prop 2`] = `
.c1 {
  margin-right: 4px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  line-height: 16px;
  color: var(--bgColor-default,var(--color-canvas-default,#ffffff));
  text-align: center;
  border-radius: 100px;
  background-color: var(--bgColor-done-emphasis,var(--color-done-emphasis,#8250df));
  color: var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));
  box-shadow: var(--boxShadow-thin,inset 0 0 0 1px) var(--borderColor-done-emphasis,transparent);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 14px;
}

<span
  className="c0"
>
  <svg
    aria-label="Issue"
    className="c1"
    fill="currentColor"
    focusable="false"
    height={16}
    role="img"
    style={
      {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  >
    <path
      d="M11.28 6.78a.75.75 0 0 0-1.06-1.06L7.25 8.69 5.78 7.22a.75.75 0 0 0-1.06 1.06l2 2a.75.75 0 0 0 1.06 0l3.5-3.5Z"
    />
    <path
      d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0Zm-1.5 0a6.5 6.5 0 1 0-13 0 6.5 6.5 0 0 0 13 0Z"
    />
  </svg>
</span>
`;

exports[`StateLabel respects the status prop 3`] = `
.c1 {
  margin-right: 4px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  line-height: 16px;
  color: var(--bgColor-default,var(--color-canvas-default,#ffffff));
  text-align: center;
  border-radius: 100px;
  background-color: var(--bgColor-neutral-emphasis,var(--color-neutral-emphasis,#6e7781));
  color: var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));
  box-shadow: var(--boxShadow-thin,inset 0 0 0 1px) var(--borderColor-neutral-emphasis,transparent);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 14px;
}

<span
  className="c0"
>
  <svg
    aria-label="Issue, not planned"
    className="c1"
    fill="currentColor"
    focusable="false"
    height={16}
    role="img"
    style={
      {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  >
    <path
      d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm9.78-2.22-5.5 5.5a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l5.5-5.5a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z"
    />
  </svg>
</span>
`;

exports[`StateLabel respects the status prop 4`] = `
.c1 {
  margin-right: 4px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  line-height: 16px;
  color: var(--bgColor-default,var(--color-canvas-default,#ffffff));
  text-align: center;
  border-radius: 100px;
  background-color: var(--bgColor-done-emphasis,var(--color-done-emphasis,#8250df));
  color: var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));
  box-shadow: var(--boxShadow-thin,inset 0 0 0 1px) var(--borderColor-done-emphasis,transparent);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 14px;
}

<span
  className="c0"
>
  <svg
    aria-label="Pull request"
    className="c1"
    fill="currentColor"
    focusable="false"
    height={16}
    role="img"
    style={
      {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  >
    <path
      d="M5.45 5.154A4.25 4.25 0 0 0 9.25 7.5h1.378a2.251 2.251 0 1 1 0 1.5H9.25A5.734 5.734 0 0 1 5 7.123v3.505a2.25 2.25 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.95-.218ZM4.25 13.5a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm8.5-4.5a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM5 3.25a.75.75 0 1 0 0 .005V3.25Z"
    />
  </svg>
</span>
`;

exports[`StateLabel respects the status prop 5`] = `
.c1 {
  margin-right: 4px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  line-height: 16px;
  color: var(--bgColor-default,var(--color-canvas-default,#ffffff));
  text-align: center;
  border-radius: 100px;
  background-color: var(--bgColor-attention-emphasis,var(--color-attention-emphasis,#9a6700));
  color: var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));
  box-shadow: var(--boxShadow-thin,inset 0 0 0 1px) var(--borderColor-attention-emphasis,transparent);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 14px;
}

<span
  className="c0"
>
  <svg
    aria-label="Pull request"
    className="c1"
    fill="currentColor"
    focusable="false"
    height={16}
    role="img"
    style={
      {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  >
    <path
      d="M3.75 4.5a1.25 1.25 0 1 0 0-2.5 1.25 1.25 0 0 0 0 2.5ZM3 7.75a.75.75 0 0 1 1.5 0v2.878a2.251 2.251 0 1 1-1.5 0Zm.75 5.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm5-7.75a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm5.75 2.5a2.25 2.25 0 1 1-4.5 0 2.25 2.25 0 0 1 4.5 0Zm-1.5 0a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z"
    />
  </svg>
</span>
`;

exports[`StateLabel respects the variant prop 1`] = `
.c1 {
  margin-right: 4px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  line-height: 16px;
  color: var(--bgColor-default,var(--color-canvas-default,#ffffff));
  text-align: center;
  border-radius: 100px;
  background-color: var(--bgColor-open-emphasis,var(--color-open-emphasis,#1f883d));
  color: var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));
  box-shadow: var(--boxShadow-thin,inset 0 0 0 1px) var(--borderColor-open-emphasis,transparent);
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
}

<span
  className="c0"
>
  <svg
    aria-label="Issue"
    className="c1"
    fill="currentColor"
    focusable="false"
    height={16}
    role="img"
    style={
      {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  >
    <path
      d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"
    />
    <path
      d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z"
    />
  </svg>
</span>
`;

exports[`StateLabel respects the variant prop 2`] = `
.c1 {
  margin-right: 4px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  line-height: 16px;
  color: var(--bgColor-default,var(--color-canvas-default,#ffffff));
  text-align: center;
  border-radius: 100px;
  background-color: var(--bgColor-open-emphasis,var(--color-open-emphasis,#1f883d));
  color: var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));
  box-shadow: var(--boxShadow-thin,inset 0 0 0 1px) var(--borderColor-open-emphasis,transparent);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 14px;
}

<span
  className="c0"
>
  <svg
    aria-label="Issue"
    className="c1"
    fill="currentColor"
    focusable="false"
    height={16}
    role="img"
    style={
      {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  >
    <path
      d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"
    />
    <path
      d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z"
    />
  </svg>
</span>
`;
